src/python/gudhi/simplex_tree.pyx

```diff
@@ -797,6 +799,37 @@ cdef class SimplexTree:
         :rtype: bool
         """
         return dereference(self.get_ptr()) == dereference(other.get_ptr())
+    
+    def __getstate__(self):
+        """Pickle the SimplexTree data structure as a Python Byte Array
+        :raises MemoryError: In the case the serialization allocates a too large block of memory.
+        :returns: Serialized SimplexTree data structure
+        :rtype: Byte Array
+        """
+        cdef char* buffer
+        cdef size_t buffer_size = 0
+        with nogil:
+            buffer = self.get_ptr().serialize(buffer_size)
+        
+        return PyByteArray_FromStringAndSize(buffer, buffer_size)
+        del buffer
```

Comment:
I don't think instructions after `return` have much effect. Isn't there a warning for unreachable code?